src/Oro/Bundle/BatchBundle/ORM/QueryBuilder/CountQueryBuilderOptimizer.php

```diff
@@ -95,7 +95,7 @@ public function getCountQueryBuilder(QueryBuilder $originalQb)
             $this->addJoins($qb, $parts);
         }
         if (!$parts['groupBy']) {
-            $fieldsToSelect[] = $this->getFieldFQN($this->idFieldName);
+            $fieldsToSelect[] = sprintf('DISTINCT %s', $this->getIdFieldFQN());
```

Comment:
I think I may be missing something as I don't see how this value could be a join or composite primary key because [here](https://github.com/LRotherfield/platform/blob/hotfix/datagrid-complex-join-pagination/src/Oro/Bundle/BatchBundle/ORM/QueryBuilder/CountQueryBuilderOptimizer.php#L185) you are making sure that only one field is returned as the primary id field?

If there is an issue then this code wont work because DISTINCT is key to this patch https://github.com/doctrine/doctrine2/blob/a41e774bb1f84d6f139438e40c45d54f042978c0/tests/Doctrine/Tests/ORM/Tools/Pagination/LimitSubqueryWalkerTest.php